In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [3]:
df = pd.read_csv("Admission_Predict.csv")

In [4]:
df.head()

Serial No.  GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  \
0           1        337          118                  4  4.5   4.5  9.65   
1           2        324          107                  4  4.0   4.5  8.87   
2           3        316          104                  3  3.0   3.5  8.00   
3           4        322          110                  3  3.5   2.5  8.67   
4           5        314          103                  2  2.0   3.0  8.21   

   Research  Chance of Admit   
0         1              0.92  
1         1              0.76  
2         1              0.72  
3         1              0.80  
4         0              0.65

In [6]:
df.columns

Index(['Serial No.', 'GRE Score', 'TOEFL Score', 'University Rating', 'SOP',
       'LOR ', 'CGPA', 'Research', 'Chance of Admit '],
      dtype='object')

In [7]:
Features = df[["GRE Score","TOEFL Score","University Rating","SOP","LOR ","CGPA","Research"]]

In [8]:
Features.head()

GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  Research
0        337          118                  4  4.5   4.5  9.65         1
1        324          107                  4  4.0   4.5  8.87         1
2        316          104                  3  3.0   3.5  8.00         1
3        322          110                  3  3.5   2.5  8.67         1
4        314          103                  2  2.0   3.0  8.21         0

In [9]:
Target = df[["Chance of Admit "]]

In [12]:
df.isnull().sum()

Serial No.           0
GRE Score            0
TOEFL Score          0
University Rating    0
SOP                  0
LOR                  0
CGPA                 0
Research             0
Chance of Admit      0
dtype: int64

In [34]:
df.corr(method ='kendall') 

Serial No.  GRE Score  TOEFL Score  University Rating  \
Serial No.           1.000000  -0.064791    -0.101776          -0.119730   
GRE Score           -0.064791   1.000000     0.667509           0.544494   
TOEFL Score         -0.101776   0.667509     1.000000           0.567294   
University Rating   -0.119730   0.544494     0.567294           1.000000   
SOP                 -0.121769   0.467137     0.514420           0.638358   
LOR                 -0.057908   0.414973     0.424169           0.547389   
CGPA                -0.033369   0.659007     0.653665           0.611896   
Research            -0.051616   0.492727     0.421512           0.411914   
Chance of Admit      0.020912   0.639933     0.625485           0.599076   

                        SOP      LOR       CGPA  Research  Chance of Admit   
Serial No.        -0.121769 -0.057908 -0.033369 -0.051616          0.020912  
GRE Score          0.467137  0.414973  0.659007  0.492727          0.639933  
TOEFL Score        0.514420  0.424169  0.653665  0.421512          0.625485  
University Rating  0.638358  0.547389  0.611896  0.411914          0.599076  
SOP                1.000000  0.600787  0.565253  0.385807          0.545497  
LOR                0.600787  1.000000  0.510758  0.350789          0.518433  
CGPA               0.565253  0.510758  1.000000  0.434767          0.720655  
Research           0.385807  0.350789  0.434767  1.000000          0.480270  
Chance of Admit    0.545497  0.518433  0.720655  0.480270          1.000000

# Linear Regression

<h1>Model by taking GRE Score, TOEFL Score, University Rating, SOP, LOR, CGPA, Research to predict the Chance of Admit

In [14]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(Features, Target, test_size=0.33, random_state=42)

In [23]:
from sklearn.linear_model import LinearRegression
reg = LinearRegression().fit(X_train, y_train)
reg.score(X_train, y_train)
print("Training Score ",reg.score(X_train,y_train))

Training Score  0.7978103892165577


In [22]:
print("Test Score ",reg.score(X_test,y_test))


Test Score  0.8044753910534395


Accurary of the model for Training Data is 0.7978103892165577 <br>
Accurary of the model for Training Data is 0.8044753910534395

<h1>Model by taking GRE Score, TOEFL Score,University Rating, SOP and CGPA to predict the Chance of Admit</h1>

In [30]:
Features_new = df[["GRE Score","TOEFL Score","University Rating","SOP","CGPA"]]
Features_new.head()

GRE Score  TOEFL Score  University Rating  SOP  CGPA
0        337          118                  4  4.5  9.65
1        324          107                  4  4.0  8.87
2        316          104                  3  3.0  8.00
3        322          110                  3  3.5  8.67
4        314          103                  2  2.0  8.21

In [31]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(Features_new, Target, test_size=0.33, random_state=42)

In [32]:
from sklearn.linear_model import LinearRegression
reg = LinearRegression().fit(X_train, y_train)
reg.score(X_train, y_train)
print("Training Score ",reg.score(X_train,y_train))

Training Score  0.7912759984131291


In [33]:
print("Test Score ",reg.score(X_test,y_test))

Test Score  0.7834612358585443


Accurary of the model for Training Data is 0.7912759984131291 <br>
Accurary of the model for Training Data is 0.7834612358585443